addons/mail/models/mail_thread.py

```diff
@@ -4045,6 +4045,7 @@ def _message_update_content(self, message, body, attachment_ids=None,
         :param str body: new body (None to skip its update);
         :param list attachment_ids: list of new attachments IDs, replacing old one (None
           to skip its update);
+        :param list attachment_ids: list of new partner IDs that are mentioned;
```

Comment:
What is the purpose of adding people that won't receive any real notification ?

Reply:
Mentions in message doesn't necessarily need to notify users.
Sure this makes a lot of sense when posting a message, but you don't want to edit a message by adding new mentions and it notifies them again. That's not how other messaging apps do (e.g. Discord/ Github), and this is a good choice to notify only on new messages.

We make mistakes, so it's good to allow editing a message. It's frustrating to not allow editing the mentions during edition, so that's why we allow having it. Handling notifying of users can be pretty hard if we put duration window (e.g. 5min.) and compare content before / after edition to notify only relevant users... But also this is a lot of complicated code that is not worth it when we can just notify on new message post and never on editing a message.

Personally I had cases I wanted to add a mention in an already posted message. Or I mistakenly removed a mention during a message edition and wanted to add it back by editing the message again. All in all, this is sometimes handy to edit mentions on a message, even when this doesn't have the side-effect to notify the newly mentioned people

Reply:
But what's the point of having a name is green like other notified people, if he was not notified ? Is it clear to users that you can mention people that won't actually receive the notification ?

Reply:
The notified part of mentions is very useful, but mentions are still valuable even without it. It shows clearly the explicit recipients of the message in the context of the content of message. For example, "`@A + question`" clearly shows you expect a question from this person. While a "cc `@B`" you just wanted to let someone know of it, without necessarily expecting a reply.
Mentions are still useful to refer to someone in a message even without the notify aspect, e.g. when posting a message "after discussion with `@A` and `@B`". If you mistakenly forgot to mention `@B` but the big message content is correct, it can be very annoying to delete and post the new message again, and have `@A` be notified again.

We were thinking on whether it made sense to notify users while editing the message, and if so when / how / etc. This can become very complicated to handle partially notifying recipients of an edited message, and this gives too much importance on editing messages rather than posting new messages. We should keep in mind that the reason editing a message exist is to allow users fixing mistakes in their own messages.

Therefore the best solution is simply to not notify the user, so that notify stays relevant for new messages and new messages only. If we want to notify someone of an old message, we simply post a new message that refers to this message.

Reply:
Also note you can click mentions, which can be useful in its own right.

I think most systems don't notify on edit so I don't mind as much as tde, people can just check the notification icon in theory. But I don't think notifying new mentions would be that difficult in python if we wanted to, you can just get the partner from each mention and call _notify_thread with `skip_existing` via a new entry point (or an existing one if it fits), and it should just notify whoever wasn't already.

Reply:
> The notified part of mentions is very useful, but mentions are still valuable even without it. It shows clearly the explicit recipients of the message in the context of the content of message. For example, "`@A + question`" clearly shows you expect a question from this person. While a "cc `@B`" you just wanted to let someone know of it, without necessarily expecting a reply. Mentions are still useful to refer to someone in a message even without the notify aspect, e.g. when posting a message "after discussion with `@A` and `@B`". If you mistakenly forgot to mention `@B` but the big message content is correct, it can be very annoying to delete and post the new message again, and have `@A` be notified again.
> 
> We were thinking on whether it made sense to notify users while editing the message, and if so when / how / etc. This can become very complicated to handle partially notifying recipients of an edited message, and this gives too much importance on editing messages rather than posting new messages. We should keep in mind that the reason editing a message exist is to allow users fixing mistakes in their own messages.
> 
> Therefore the best solution is simply to not notify the user, so that notify stays relevant for new messages and new messages only. If we want to notify someone of an old message, we simply post a new message that refers to this message.

Ok, makes sense. I don't necessarily agree on all points but makes sense :) note that this is the really interesting point that should be in commit message, explaining why we to do or don't do something (hence asking regularly to improve messages to avoid side discussions on PRs ^^ ).